src/battle/other_battle_calculators.c

```diff
@@ -84,6 +84,160 @@ const u16 TriageMovesList[] = {
     MOVE_WISH,
 };
 
+// List of multi-strike moves
+// (https://bulbapedia.bulbagarden.net/wiki/Multi-strike_move)
+const u16 MultiHitMovesList[] = {
+    // Variable number of strikes
+    MOVE_ARM_THRUST,
+    MOVE_BARRAGE,
+    MOVE_BONE_RUSH,
+    MOVE_BULLET_SEED,
+    MOVE_COMET_PUNCH,
+    MOVE_DOUBLE_SLAP,
+    MOVE_FURY_ATTACK,
+    MOVE_FURY_SWIPES,
+    MOVE_ICICLE_SPEAR,
+    MOVE_PIN_MISSILE,
+    MOVE_ROCK_BLAST,
+    MOVE_SCALE_SHOT,
+    MOVE_SPIKE_CANNON,
+    MOVE_TAIL_SLAP,
+    MOVE_WATER_SHURIKEN,
+    // Fixed number of multiple strikes
+    MOVE_BONEMERANG,
+    MOVE_DOUBLE_HIT,
+    MOVE_DOUBLE_IRON_BASH,
+    MOVE_DOUBLE_KICK,
+    MOVE_DRAGON_DARTS,
+    MOVE_DUAL_CHOP,
+    MOVE_DUAL_WINGBEAT,
+    MOVE_GEAR_GRIND,
+    MOVE_SURGING_STRIKES,
+    MOVE_TRIPLE_DIVE,
+    MOVE_TWIN_BEAM,
+    MOVE_TWINEEDLE,
+    // Accuracy-dependent multiple strikes
+    MOVE_TRIPLE_AXEL,
+    MOVE_TRIPLE_KICK,
+    MOVE_POPULATION_BOMB,
+    // Party-dependent multiple strikes
+    MOVE_BEAT_UP,
+};
+
+// List of moves that should not hit twice when user has Parental Bond
+// (https://bulbapedia.bulbagarden.net/wiki/Parental_Bond_(Ability))
+const u16 ParentalBondSingleStrikeMovesList[] = {
+    // One-hit knockout moves
+    MOVE_FISSURE,
+    MOVE_GUILLOTINE,
+    MOVE_HORN_DRILL,
+    MOVE_SHEER_COLD,
+    // No category
+    MOVE_FLING,
+    MOVE_SELF_DESTRUCT,
+    MOVE_EXPLOSION,
+    MOVE_FINAL_GAMBIT,
+    MOVE_UPROAR,
+    MOVE_ROLLOUT,
+    MOVE_ICE_BALL,
+    MOVE_ENDEAVOR,
+    // Moves with a charging turn
+    MOVE_BOUNCE,
+    MOVE_DIG,
+    MOVE_DIVE,
+    // MOVE_ELECTRO_SHOT, // Not implemented yet
+    MOVE_FLY,
+    MOVE_FREEZE_SHOCK,
+    MOVE_GEOMANCY,
+    MOVE_ICE_BURN,
+    MOVE_METEOR_BEAM,
+    MOVE_PHANTOM_FORCE,
+    MOVE_RAZOR_WIND,
+    MOVE_SHADOW_FORCE,
+    MOVE_SKULL_BASH,
+    MOVE_SKY_ATTACK,
+    MOVE_SKY_DROP,
+    MOVE_SOLAR_BEAM,
+    MOVE_SOLAR_BLADE,
+    // Z-Moves
+    MOVE_BREAKNECK_BLITZ_PHYSICAL,
```

Comment:
may at some point just move all these tables with z moves and gmax stuff to different functions to save space